### PR TITLE
Enhance analytics

### PR DESCRIPTION
**Pull request by mattystank**
**Does not cover:**

-Direct
-Global Tabs (need to fix routing)
-Details in scorecard & validators

### CODE REVIEW

File: src/components/c-cda/validation/results/ValidationResultsDialog.tsx [hunk 1]
```diff
@@ -33,6 +33,16 @@ const ScrollableDialog: React.FC<ScrollableDialogProps> = ({ open, handleClose,
   const handlePrint = useReactToPrint({
     content: () => contentRef.current,
   })
+  const handleButtonClick = () => {
```
Comment (drbgfc) [inline src/components/c-cda/validation/results/ValidationResultsDialog.tsx:36]: Suggest we call this function something like trackSaveButtonClick instead for 2 reasons:
1: We have other buttons in the file, so this name is a bit generic and may need to change in the future
2: Consistency with other GA4 code in the commit

**Review by drbgfc — CHANGES_REQUESTED** [review]
The code looks good and has a good coverage of events.  I have a limited understanding of GA4, but at face value I think we have 3 choices here and just ask that figure out the best path and make it consistent with regards to naming and impl.

1: Share one onClick function for functionality and tracking
--This is not great for separation of concerns, but, it is quicker to implement. We have done this in most cases here. Maybe we should do all of them that way for consistency, if choosing this route?

2: Use a separate onClick function for functionality and GA4
--This separates concerns. This would be best practice in my mind, but, does require more code. It also improves the performance of the actual functionality

* 3: Use onClick function for functionality and onClickCapture for tracking GA4
--This has all the advantages of option 2, and offers even more separation of concerns/clarity on intent. It also ensures the event is tracked whether or not the functionality worked/has no side effects that can interfere with the tracking. For these reasons, this is my favorite, suggested approach. However, it's up to you if you feel like adjusting to that approach as it requires the most changes.

**Comment by mattystank** [comment]
Lets go with option 3. However I would to see this merge and we clean up later? 1 to check if analytics are coming thru & 2. I want to share you the report in GA4 and see if we need to adjust then... I know there will be other analytics tickets in the near future.

> 3: Use onClick function for functionality and onClickCapture for tracking GA4 --This has all the advantages of option 2, and offers even more separation of concerns/clarity on intent. It also ensures the event is tracked whether or not the functionality worked/has no side effects that can interfere with the tracking. For these reasons, this is my favorite, suggested approach. However, it's up to you if you feel like adjusting to that approach as it requires the most changes.

**Comment by drbgfc** [comment]
> Lets go with option 3. However I would to see this merge and we clean up later? 1 to check if analytics are coming thru & 2. I want to share you the report in GA4 and see if we need to adjust then... I know there will be other analytics tickets in the near future.
> 
> > 3: Use onClick function for functionality and onClickCapture for tracking GA4 --This has all the advantages of option 2, and offers even more separation of concerns/clarity on intent. It also ensures the event is tracked whether or not the functionality worked/has no side effects that can interfere with the tracking. For these reasons, this is my favorite, suggested approach. However, it's up to you if you feel like adjusting to that approach as it requires the most changes.

Sounds good 👍

**Pull request merged**